R/trackhub.R

```diff
@@ -0,0 +1,69 @@
+### =========================================================================
+### TrackHub support
+### -------------------------------------------------------------------------
+
+### - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+### TrackHub class
+###
+
+setClass("TrackHub", representation(uri = "character"))
+
+isFileReference <- function(s) {
+  length(grep(".txt", s)) == 1
+}
+
+uri <- function(x, ...) x@uri
+
+getContent <- function(x) {
```

Comment:
Consider using just a named character vector for these attributes. Simpler than a data.frame, and there would be no need for `getValue()`, as values would be accessed like `hubContent["genomesFile"]`.